finality-aleph/src/data_io/status_provider.rs

```diff
@@ -238,7 +248,7 @@ mod tests {
                 cached_cip,
                 aux_cip,
                 &proposal,
-                ProposalStatus::Finalize(proposal.top_block()),
+                ProposalStatus::Finalize(proposal.blocks_from_num(0).collect()),
```

Comment:
Would be great to have even a single test that checks the case when only some part of the branch is supposed to be finalized